pkg/webhook/validation/validate_service.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2020 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"knative.dev/pkg/logging"
+)
+
+// ExtraServiceValidation runs extra validation on Service resources
+func ExtraServiceValidation(ctx context.Context, uns *unstructured.Unstructured) error {
+	content := uns.UnstructuredContent()
+
+	// TODO(whaught): remove this guard once variations of this are well-tested
+	// Only run extra validation for the dry-run test. This will be in place to while
+	// the feature is tested for compatibility and later removed.
+	m, found, err := unstructured.NestedMap(content, "metadata", "annotations")
+	if !found || err != nil {
+		return nil
+	}
+	testAnn := m["knative-e2e-test"]
+	if testAnn != "TestServiceValidationWithInvalidPodSpec" {
```

Comment:
I don't want to hard code test names since this can easily break if the test name changes and some future developer may not understand why.

Discussion in slack:
https://knative.slack.com/archives/CA4DNJ9A4/p1587071192305600